### ISSUE TITLE
exFAT boot support?

### ISSUE THREAD

**Issue by ZhuMa12345678** [maharmstone/quibble#15]
Many years ago, I installed and started Windows on the exFAT partition, but only supported Windows 8 and above, I hope this project can support exFAT to boot Windows 7 and below.
 
exFAT boot preview
![image](https://user-images.githubusercontent.com/61312109/75026320-bcc5c180-54d7-11ea-8514-c84228850086.png)
 
This is a detailed installation tutorial I wrote in Simplified Chinese (because I am from China)
http://bbs.wuyou.net/forum.php?mod=viewthread&tid=388226&mobile=no
I appologize for my grammatical mistake because English isn't my main language.

**Comment by maharmstone** [comment]
Well, you've already proven that the Windows exfat driver is good enough to boot off. All you need to do is find an exfat UEFI driver that's good enough - have you tried https://github.com/pbatard/efifs?

**Comment by ZhuMa12345678** [comment]
exFAT boot Windows system is supported in both Legacy and UEFI boot modes.
Yes, I use the exfat.efi driver of EfiFs without any problems.
I also converted exfat.efi to exfat.ffs and built it into the UEFI BIOS as a DXE driver, so that the UEFI firmware can get rid of the dependence on the FAT/32 file system.
![image](https://user-images.githubusercontent.com/61312109/75089821-b4cd5680-5597-11ea-80df-a43df44a04cc.png)
exFAT UEFI boot preview screenshot
![image](https://user-images.githubusercontent.com/61312109/75089827-b8f97400-5597-11ea-99fd-1dcc00e0900b.png)
This is a detailed tutorial I wrote:
http://bbs.wuyou.net/forum.php?mod=viewthread&tid=409768&mobile=no
I appologize for my grammatical mistake because English isn't my main language.

**Comment by MegaZeroX10** [comment]
Can you please make an English version of the guide?

**Comment by ZhuMa12345678** [comment]
@MegaZeroX10 
Sorry, I don't have a professional human translator. I can only use machine translation, but this will cause some translation problems in this guide.

**Comment by ZhuMa12345678** [comment]
@maharmstone 
Hi, I tried to put exFAT.efi in the drivers folder, and then when I booted quibble.efi, it showed that the exFAT driver was loaded successfully.
![image](https://user-images.githubusercontent.com/61312109/81425796-d5667f00-918a-11ea-8979-e7dd55d6b63e.png)
However, when booting Windows, an error was reported, it seems that quibble cannot read system files in the exFAT partition.
![image](https://user-images.githubusercontent.com/61312109/81425803-d8fa0600-918a-11ea-9851-7c7b10830430.png)

**Comment by maharmstone** [comment]
Is this using the Batard driver? `EFI_NO_MAPPING` is not an error I've ever used anywhere (I don't think), it comes from the driver. Judging from the search results (https://github.com/pbatard/efifs/search?q=EFI_NO_MAPPING&unscoped_q=EFI_NO_MAPPING), I wonder if the problem might be caused by it misinterpreting a Chinese character set as UTF-8.

Could you try something for me please? In Windows, rename the System32 directory so it's all lower case, "system32". Does that change the error you receive?

**Comment by ZhuMa12345678** [comment]
Yes, I use the Batard driver. Indeed, when I renamed "System32" to "system32", booting quibble again will not report an error, and now the system files in the exFAT partition can be read correctly.
![image](https://user-images.githubusercontent.com/61312109/81428287-a6520c80-918e-11ea-8ba2-de969220ac4c.png)
But I don't quite understand why under the same conditions, this problem does not exist in NTFS booting, but this problem occurs in exFAT booting? Is there a better solution to this problem?

**Comment by maharmstone** [comment]
That'll be because NTFS stores all filenames as UTF-16, whereas it looks like ExFAT uses your native codepage.

Thanks for reporting this - I'll try to get it sorted for the next version.

**Comment by ZhuMa12345678** [comment]
@maharmstone 
Hi, I get an error when I try to boot with VMware Workstation 15 Pro.
![image](https://user-images.githubusercontent.com/61312109/81494341-c2fb5b00-92da-11ea-8046-d3558662ab10.png)
I remember that VMware seems to support CSM, do I still need to add some DXE drivers to its UEFI firmware?
In addition, I also tried to use QEMU emulator version 4.2 to load the precompiled version OVMF you provided, but QEMU reported an error and could not load the file, What is the reason?
![image](https://user-images.githubusercontent.com/61312109/81494351-cdb5f000-92da-11ea-90f2-8166b4d7a21d.png)

**Comment by a1ive** [comment]
> @maharmstone
> Hi, I get an error when I try to boot with VMware Workstation 15 Pro.
> ![image](https://user-images.githubusercontent.com/61312109/81494341-c2fb5b00-92da-11ea-8046-d3558662ab10.png)
> I remember that VMware seems to support CSM, do I still need to add some DXE drivers to its UEFI firmware?
> In addition, I also tried to use QEMU emulator version 4.2 to load the precompiled version OVMF you provided, but QEMU reported an error and could not load the file, What is the reason?
> ![image](https://user-images.githubusercontent.com/61312109/81494351-cdb5f000-92da-11ea-90f2-8166b4d7a21d.png)

-bios xxx.fd => -pflash xxx.fd

**Comment by ZhuMa12345678** [comment]
Thanks for reminding, that the QEMU emulator can now be turned on. But I encountered a new problem, as shown in this animation, it will restart infinitely when loading the CSM stage.
![Test](https://user-images.githubusercontent.com/61312109/81495711-c398ef00-92e4-11ea-94ad-106824ef71f0.gif)
